Plutarch/Api/V1/Tx.hs

```diff
@@ -38,6 +43,24 @@ instance DerivePlutusType PTxId where type DPTStrat _ = PlutusTypeData
 instance PUnsafeLiftDecl PTxId where type PLifted PTxId = Plutus.TxId
 deriving via (DerivePConstantViaData Plutus.TxId PTxId) instance PConstantDecl Plutus.TxId
 
+instance PTryFrom PData PTxId where
+  type PTryFromExcess PData PTxId = Flip Term PTxId
+  ptryFrom' opq cont = ptryFrom @(PAsData PTxId) opq (cont . first punsafeCoerce)
+
+instance PTryFrom PData (PAsData PTxId) where
+  type PTryFromExcess PData (PAsData PTxId) = Flip Term PTxId
+  ptryFrom' opq = runTermCont $ do
+    opq' <- tcont . plet $ pasConstr # opq
+    dataBs <- tcont $ \f ->
+      pif
+        (pfstBuiltin # opq' #== 0 #&& plength # (psndBuiltin # opq') #== 1)
+        (f $ phead #$ psndBuiltin # opq')
+        (ptraceError "ptryFrom(TxId): bad constructor")
+    unwrapped <- tcont . plet $ ptryFrom @(PAsData PByteString) dataBs snd
+    tcont $ \f ->
+      pif (plengthBS # unwrapped #== 28) (f ()) (ptraceError "ptryFrom(TxId): must be 28 bytes long")
+    pure (punsafeCoerce opq, pcon . PTxId $ pdcons # pdata unwrapped # pdnil)
```

Comment:
@L-as Is this the most ergonomic way to implement `PTryFrom` for data sum stuff currently? Surely users shouldn't be expected to perform structure validation when the target invariant is only for one of the fields within - I assume there is a better solution utilizing `PTryFromExcess`?

Reply:
From my other comment:
>  Optimally you should be able to first verify it's a PDataSum, but because it doesn't return any excess, it's not efficient.